engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/ExecutableMetaData.java

```diff
@@ -307,14 +310,35 @@ public boolean equals(Object obj) {
 		 * @param constraintHelper the constraint helper
 		 * @param executableHelper the executable helper
 		 * @param beanClass the bean class
+		 * @param methodValidationConfiguration configuration instance for method validation behaviour
 		 */
-		public Builder(Class<?> beanClass, ConstrainedExecutable constrainedExecutable, ConstraintHelper constraintHelper, ExecutableHelper executableHelper) {
+		public Builder(
+				Class<?> beanClass,
+				ConstrainedExecutable constrainedExecutable,
+				ConstraintHelper constraintHelper,
+				ExecutableHelper executableHelper,
+				MethodValidationConfiguration methodValidationConfiguration) {
 			super( beanClass, constraintHelper );
 
 			this.executableHelper = executableHelper;
 			kind = constrainedExecutable.getKind();
 			executable = constrainedExecutable.getExecutable();
 			add( constrainedExecutable );
+			this.methodValidationConfiguration = methodValidationConfiguration;
+
+			// Build the rules that will be enforced through the metadata created here
+			this.rules = new HashSet<MethodConfigurationRule>();
+			for ( Class<? extends MethodConfigurationRule> ruleClass : this.methodValidationConfiguration.getConfiguredRuleSet() ) {
+				try {
+					this.rules.add( ruleClass.newInstance() );
```

Comment:
Please use `org.hibernate.validator.internal.util.privilegedactions.NewInstance` instead; It may not be strictly needed in this particular case, but I'd prefer it for the sake of consistency throughout the code base (also exception messages will be consistent that way).

Or, couldn't we have the actual rule *instances*  in this set already (as created earlier in `MethodValidationConfigurationImpl()`? That'd avoid the need for reflective instantiation altogether.